lib/tort.rb

```diff
@@ -1,3 +1,26 @@
-# stub class for development
+require 'parallel'
+
+# Multi-threaded / multi-processed sort for Ruby
 class Tort
+  def self.tort_thread_sort(unsorted_array, thread_workers = Parallel.processor_count, &block)
+    Parallel.map(chunk_array(unsorted_array, thread_workers), in_threads: thread_workers) do |sub_array|
+      sub_array.sort(&block)
+    end.inject(&method(:merge))
+  end
+
+  def self.tort_process_sort(unsorted_array, process_workers = Parallel.processor_count, &block)
+    Parallel.map(chunk_array(unsorted_array, process_workers), in_processes: process_workers) do |sub_array|
+      sub_array.sort(&block)
+    end.inject(&method(:merge))
+  end
+
```

Comment:
Should chunk_array and merge be exposed? If they have no use outside of thread sort and process sort then it would make sense to change them to private, non-static methods.

Reply:
You can use private_class_method :chunk_array, :merge for this.

Reply:
OOoooooooOOOoOoOoOOoooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo

I'll do this!